dev-docs/howtoguides/release_a_new_version.md

```diff
@@ -39,174 +49,212 @@ If this is your first time releasing ubuntu-advantage-tools, you'll need to do t
+  commit id: "v42-feature2" tag: "42-rc"
+
+  branch next-v43 order: 3
+  commit id: "v43-feature1"
+  commit id: "v43-feature2"
+  checkout main
+  commit id: "v42-fixup1"
+  checkout next-v43
+  commit id: "v43-feature3"
+  checkout main
+  commit id: "v42-fixup2" tag: "42"
+  checkout review-v42
+  merge main id: "Release PR" type:HIGHLIGHT
+  checkout main
+  merge next-v43
+```
+
+`main` is the only long-lived branch.
+
+When we reach a point in `main` where we are ready to release version 42, we do the following:
+* Tag that commit as `42-rc`
+* Create a branch called `review-v42` from the `41` tag
+  * This branch does not have any new commits on it, it is only used as a target for the PR that will be created in the next step
+* Create a draft PR in GitHub from `main` targeting `review-v42`
+* Create a branch called `next-v43` from the `42-rc` tag
+* For any open PRs that are not to be included in version 42, change their target to `next-v43`
+  * Any new PRs for ongoing development during the version 42 release process must target `next-v43`
+* Sponsor and SRU review uses the `main`->`review-v42` PR
+  * Feedback that requires changes are added to `main`
+* When version 42 is approved, the tip of `main` is tagged `42` and the Sponsor uploads using that commit
+  * The `review-v42` branch is now deleted along with the `main`->`review-v42` PR (the PR is not merged).
+* After version 42 is released, `next-v43` is rebased on top of `main` and `next-v43` is deleted
+* Development continues on `main` until we are ready to begin releasing version 43
+
+## Release process
+
+The process will use `$version` throughout in place of the version number, e.g., `42`. The process will also use `$devel_release` in place of the current devel release, e.g., `noble`.
+
+### 1. Set up `review` and `next` branches
+
+```bash
+# Make sure you are up to date
+git switch main
+git pull
+
+# Tag the release candidate
+git tag $version-rc # e.g., 42-rc
+git push origin $version-rc
+
+# Create the next branch
+git switch -c next-v$((version+1))
+git push origin next-v$((version+1))
+
+# Create the review branch
+git switch main
+git switch -c review-v$version
+git push origin review-v$version
+```
+
+Now configure two things:
+1. Change GitHub's "default" branch to `next-v$((version+1))`
+    - This should automatically switch any open PRs against `main` to target `next-v$((version+1))`.
+2. Change the daily PPA build recipe to use `next-v$((version+1))`
+
+And finally, create a draft PR from `main` to `review-v$version` on GitHub.
+
+### 2. Write a changelog entry
+
+Create a PR against `main` finishing the changelog entry for the new version. There should be an entry at the top for `UNRELEASED` with the version `1:1+devel`. Change the version to `$version` and change `UNRELEASED` to `$devel_release`. Also review the contents of the changelog entry and add anything that is missing. This PR should be reviewed and merged before starting the release process.
```

Comment:
Why should we keep in main the `1:1+devel` entry in the changelog ? Shouldn't this entry only exist in the `next` branch ?

Reply:
(We discussed this in a meeting): the language here is a bit confusing. This is saying that you'll have to change the current 1:1+devel that is already there into the real release changelog entry.

`main` will have 1:1+devel during development because the daily ppa will be building from it.

```suggestion
Create a PR against `main` finishing the changelog entry for the new version. In the top entry, change `1:1+devel` to `$version` and change `UNRELEASED` to `$devel_release`. Also review the contents of the changelog entry and add anything that is missing. This PR should be reviewed and merged before starting the release process.
```